src/FieldWarning/Assets/UI/Ingame/UnitLabel/WaypointOverlayBehavior.cs

```diff
@@ -0,0 +1,49 @@
+﻿using PFW.Units;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class WaypointOverlayBehavior : MonoBehaviour
+{
+
+    private PlatoonBehaviour _platoon;
+    private LineRenderer _lineR;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (_platoon != null)
```

Comment:
I think this null check is a perfect example for the bug hiding thing. We have the bug of waypoints showing after the platoon has been destroyed. If this null check wasn't here, you'd immediately get an error in that case, but instead the code just quietly fails and you're left wondering why.